src/solver/impls/arkode/arkode.cxx

```diff
@@ -226,7 +226,7 @@ ArkodeSolver::ArkodeSolver(Options* opts)
                        .withDefault(false)),
       optimize(
           (*options)["optimize"].doc("Use ARKode optimal parameters").withDefault(false)),
-      suncontext(MPI_COMM_WORLD) {
+      suncontext(static_cast<void*>(MPI_COMM_WORLD)) {
```

Comment:
This isn't quite the right fix, I'm afraid. For MPICH, `MPI_COMM_WORLD` is an `int` literal, so this is casting that integer value to a random address. We actually need to pass a pointer to the communicator, so `static_cast<void*>&BoutComm::get()`